constellations/orion/src/controllers/classes_by_semesters_controller.go

```diff
@@ -0,0 +1,125 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/gin-gonic/gin"
+)
+
+func GetAllProgramsSemestersClasses(c *gin.Context) {
+	var classSemester domains.ProgramClassesBySemester
```

Comment:
I'm not a fan of having this variable here. It's only used in the for loop so this variable should be defined only inside the for loop.
If it is defined outside the for loop, it tells me that it might be used OUTside the for loop which it never is.